app/src/main/java/net/aiscope/gdd_app/ui/metadata/MetadataMapper.kt

```diff
@@ -0,0 +1,80 @@
+package net.aiscope.gdd_app.ui.metadata
+
+import android.content.Context
+import android.content.res.Resources
+import net.aiscope.gdd_app.R
+import net.aiscope.gdd_app.model.MalariaSpecies
+import net.aiscope.gdd_app.model.MalariaStage
+import net.aiscope.gdd_app.model.SmearType
+
+class MetadataMapper {
```

Comment:
Like I was saying above, it's great that this logic is extracted here. Do you think that it could also be unit-tested?

Reply:
That makes sense. Commit 1570e0fa414e42ef8f3280e75b6d8fb34fe28085 has a new MetadataMapperTest class with tests. Can you verify that this is the best way to write the tests?

Reply:
They look good to me, I don't think that there's a better way due to those `Context.getString()` calls. The only thing I see is that test rule for coroutines, which is not used.